fastlane_bot/helpers/txhelpers.py

```diff
@@ -775,31 +789,22 @@ def approve_token_for_arb_contract(self, token_address: str, approval_amount: in
         max_priority = int(self.get_max_priority_fee_per_gas_alchemy()) if self.ConfigObj.NETWORK in ["ethereum", "coinbase_base"] else 0
 
         token_contract = self.web3.eth.contract(address=token_address, abi=ERC20_ABI)
-        try:
-            approve_tx = token_contract.functions.approve(self.arb_contract.address, approval_amount).build_transaction(
-                    self.build_tx(
-                        base_gas_price=current_gas_price, max_priority_fee=max_priority, nonce=self.get_nonce()
-                    )
-                )
-        except Exception as e:
-            self.ConfigObj.logger.info(f"Error when building transaction: {e.__class__.__name__} {e}")
-            if "max fee per gas less than block base fee" in str(e):
-                try:
-                    message = str(e)
-                    baseFee = int_prefix(message.split("baseFee: ")[1])
-                    approve_tx = token_contract.functions.approve(self.arb_contract.address,
-                                                                  approval_amount).build_transaction(
-                        self.build_tx(
-                            base_gas_price=baseFee, max_priority_fee=max_priority, nonce=self.get_nonce()
-                        )
-                    )
-                    self.ConfigObj.logger.info(f"Submitting approval for token: {token_address}")
-                    return self.submit_regular_transaction(self.sign_transaction(approve_tx))
-                except Exception as e:
-                    self.ConfigObj.logger.info(
-                        f"(***2***) Error when building transaction: {e.__class__.__name__} {e}")
-            else:
-                return None
+
+
+        approve_tx = self.build_transaction_generic(
```

Comment:
As far as I'm aware of, there is no function `build_transaction_generic` anywhere in the code.

Reply:
Look again